### PR TITLE
[KARAF-6579] Add Commands jaas:realm-add and jaas:module-add to manag…

### PR DESCRIPTION
**Pull request by JulianFeinauer**
…e realms and modules via shell

See the documentation added to security.adoc which describes all changes in detail.

### CODE REVIEW

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package org.apache.karaf.jaas.command;
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java]: ASF header is missing.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+public class ModuleAddCommand extends JaasCommandSupport {
+
+    @Argument(index = 0, name = "loginModule", description = "Class Name of Login Module", required = true, multiValued = false)
+    private String loginModule;
+
+    @Argument(index = 1, name = "properties", description = "Pair of Properties (key value)", required = false, multiValued = true)
+    private List<String> propertiesList;
+
+    @Override
+    protected Object doExecute(BackingEngine engine) throws Exception {
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java]: Why extending `JaasCommandSupport` if we don't use the method ? I would directly implement `Action`.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            return null;
+        }
+        if (!(realm instanceof Config)) {
+            System.err.println("Selected JAAS Realm was not added via jaas:add-realm, only those are supported!");
+            return null;
+        }
+
+        if (!checkIfClassExists(loginModule)) {
+            System.err.println("Module class '" + loginModule + "' is unknown!");
+            return null;
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java]: We could raise exception there.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    @Override
+    public Object execute() throws Exception {
+        // Fetch Realm
+        JaasRealm realm = (JaasRealm) session.get(JAAS_REALM);
+
+        if (realm == null) {
+            System.err.println("No JAAS Realm has been selected");
+            return null;
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java]: We can throw an `IllegalArgumentException` there.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        }
+        if (!(realm instanceof Config)) {
+            System.err.println("Selected JAAS Realm was not added via jaas:add-realm, only those are supported!");
+            return null;
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/ModuleAddCommand.java]: Same here.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/RealmAddCommand.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
+package org.apache.karaf.jaas.command;
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/RealmAddCommand.java]: No ASF header.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/RealmAddCommand.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+@Command(scope = "jaas", name = "realm-add", description = "Add a realm")
+@Service
+public class RealmAddCommand extends JaasCommandSupport {
+
+    @Reference
+    private BundleContext context;
+
+    @Argument(index = 0, name = "realmname", description = "Realm Name", required = true, multiValued = false)
+    private String realmname;
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/RealmAddCommand.java]: Could be cool to add completer.

File: jaas/command/src/main/java/org/apache/karaf/jaas/command/RealmAddCommand.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Argument(index = 1, name = "loginModule", description = "Class Name of Login Module", required = true, multiValued = false)
```
Comment (jbonofre) [inline jaas/command/src/main/java/org/apache/karaf/jaas/command/RealmAddCommand.java]: Same here for completer.

**Comment by jbonofre** [comment]
Generally speaking, I would go via a `ManagedService`. The command can create the configuration. But go ahead this way, we will improve in a second step.

**Comment by JulianFeinauer** [comment]
@jbonofre I adressed all your issues : )

**Comment by fpapon** [comment]
Sounds good to me

**Comment by jbonofre** [comment]
Can you please fix the test ?

**Comment by JulianFeinauer** [comment]
@jbonofre ofc, just saw it. I just pushed the fixed test now.

**Comment by jbonofre** [comment]
Test failure is not related to the PR. I'm doing a final pass.

**Comment by JulianFeinauer** [comment]
@jbonofre thanks for the support, I also created an Issue for the failing test https://issues.apache.org/jira/browse/KARAF-6590

**Pull request merged**